my-app/src/app/app.component.ts

```diff
@@ -1,10 +1,32 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
+import { fromEvent, Subscription } from 'rxjs';
+import { ContextStoreFacadeService } from './store/context-store/services/store-facade.service';
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.css']
 })
-export class AppComponent {
-  title = 'my-app';
+export class AppComponent implements OnInit {
+  public title = 'my-app';
+
+  private subscriptions: Subscription = new Subscription();
+
+  constructor(private contextStoreFacadeService: ContextStoreFacadeService) {}
+
+  public ngOnInit(): void {
+    this.contextStoreFacadeService.dispatchFetchCoursesList();
+
+    this.subscriptions.add(
```

Comment:
Don't forget to unsubscribe on component destroy. 
Also, prefer HostListener as it manages attaching and detaching of the listener itself
@HostListener('window:online')

Reply:
fixed